app/src/main/java/com/github/leuludyha/ibdb/presentation/components/books/ratings/Rating.kt

```diff
@@ -0,0 +1,152 @@
+    }
+}
+
+@Composable
+private fun RatingMedium(
+    value: Float,
+    onChange: ((Float) -> Unit)? = null,
+    label: String? = null
+) {
+    Column(
+        verticalArrangement = Arrangement.Center,
+        horizontalAlignment = Alignment.CenterHorizontally,
+    ) {
+        label?.let {
+            Text(
+                text = label,
+                style = MaterialTheme.typography.titleMedium
+            )
+        }
+        RatingSmall(value = value, onChange = onChange)
+        Text(
+            text = "${"%.1f".format(value)}/5",
+            style = MaterialTheme.typography.titleSmall
+        )
+    }
+}
+
+@Composable
+private fun RatingSmall(
+    value: Float,
+    onChange: ((Float) -> Unit)? = null,
+) {
+    Row(
+        verticalAlignment = Alignment.CenterVertically,
+        horizontalArrangement = Arrangement.Center
+    ) {
+        (0 until 5).map { i ->
+            // Collects the size of the background when it is mounted on the view
+            val (iconSize, setIconSize) = remember { mutableStateOf(Pair(0f, 0f)) }
+
+            val factor = clamp(value - i, 0f, 1f)
+
+            val onChangeI: (() -> Unit)? = if (onChange == null) null else {
+                { onChange((i + 1).toFloat()) }
+            }
+
+            val gradient = Brush.linearGradient(
+                colors = listOf(
+                    MaterialTheme.colorScheme.primary,
+                    MaterialTheme.colorScheme.scrim,
+                ),
+                start = Offset(factor * iconSize.first, 0f),
+                end = Offset(factor * iconSize.first + .1f, 0f),
+                tileMode = TileMode.Clamp
+            )
+
+            Icon(
+                modifier = Modifier
+                    .graphicsLayer(alpha = 0.99f)
+                    .optionalClick(onChangeI)
+                    .onGloballyPositioned {
+                        // Save background size when mounted
+                        setIconSize(Pair(it.size.width.toFloat(), it.size.height.toFloat()))
+                    }
+                    .drawWithCache {
+                        onDrawWithContent {
+                            drawContent()
+                            drawRect(gradient, blendMode = BlendMode.SrcAtop)
+                        }
+                    },
+                imageVector = Icons.Default.StarRate,
+                contentDescription = null
+            )
+        }
+    }
+}
+
+/**
+ * Helper function to remove visual effects in case
+ * the click cannot be handled
+ */
+private fun Modifier.optionalClick(
+    onChange: (() -> Unit)? = null,
+): Modifier {
+    if (onChange == null) {
+        return this; }
+    return this.clickable { onChange() }
+}
+
+@Preview
+@Composable
+private fun DefaultPreview() {
+    IBDBTheme {
+        Rating(
+            value = 3.5f,
+            onChange = { Log.i("Rating", it.toString()) },
+            size = RatingSize.Small,
+            label = "My Rating"
+        )
+    }
+}
```

Comment:
We are greatly struggling to get 80 percent coverage, the default previews lower it considerably. Maybe we could remove them from the pulls. I know it is annoying, but still better than falling below the threshold.